GenderPayGap.WebUI/Helpers/ReportFiguresHelper.cs

```diff
@@ -298,11 +298,32 @@ private static void ValidateHourlyPayFigures(ReportFiguresViewModel viewModel, H
         private static void ParseAndValidateParameters(
             ReportFiguresViewModel viewModel, 
             HttpRequest request, 
-            Expression<Func<ReportFiguresViewModel, decimal?>> propertyLambdaExpressions)
+            Expression<Func<ReportFiguresViewModel, decimal?>> propertyLambdaExpression)
         {
-            if (!viewModel.HasErrorFor(propertyLambdaExpressions))
+            ValidateParameterIfNoErrors(
+                viewModel,
+                propertyLambdaExpression,
+                () => viewModel.ParseAndValidateParameters(request, propertyLambdaExpression));
+        }
+        private static void AddErrorFor(
+            ReportFiguresViewModel viewModel, 
+            Expression<Func<ReportFiguresViewModel, decimal?>> propertyLambdaExpression,
+            string errorMessage)
+        {
+            ValidateParameterIfNoErrors(
+                viewModel,
+                propertyLambdaExpression,
+                () => viewModel.AddErrorFor(propertyLambdaExpression, errorMessage));
+        }
+
+        private static void ValidateParameterIfNoErrors(
+            ReportFiguresViewModel viewModel, 
+            Expression<Func<ReportFiguresViewModel, decimal?>> propertyLambdaExpression,
+            Action callback)
```

Comment:
We could give this a more descriptive name like `validatorFunction` or `validatorCallback`

Reply:
Also is there a reason we use `Action` and not `Func`?

Reply:
agreed 👍